src/v2/mod.rs

```diff
@@ -142,6 +142,21 @@ impl Client {
     }
 }
 
+/// Reqwest related implementation
+impl Client {
+    /// Takes reqwest's async RequestBuilder and injects an authentication header if a token is present
+    fn build_reqwest(
```

Comment:
Let's have this in the already existing `impl Client` block above, as it will eventually be moved there anyway.